src/main/java/com/sinthoras/visualprospecting/mixins/ifu/ItemOreFinderToolMixin.java

```diff
@@ -0,0 +1,64 @@
+package com.sinthoras.visualprospecting.mixins.ifu;
+
+import com.encraft.dz.items.*;
+import com.sinthoras.visualprospecting.*;
+import com.sinthoras.visualprospecting.database.*;
+import gregtech.api.enums.*;
+import gregtech.api.objects.*;
+import java.util.*;
+import java.util.stream.*;
+import net.minecraft.block.*;
+import net.minecraft.entity.*;
+import net.minecraft.entity.player.*;
+import net.minecraft.item.*;
+import net.minecraft.world.*;
+import org.spongepowered.asm.mixin.*;
+import org.spongepowered.asm.mixin.injection.*;
+import org.spongepowered.asm.mixin.injection.At.*;
+import org.spongepowered.asm.mixin.injection.callback.*;
+
+@Mixin(value = ItemOreFinderTool.class, remap = false)
+public class ItemOreFinderToolMixin extends Item {
+
+  @Shadow
+  private static int found;
+  
+  @Shadow
+  private static int MAX_DAMAGE;
+
+  @Inject(
+      method = "onUpdate",
+      at = @At(value = "INVOKE", target = "Lcom/encraft/dz/items/ItemOreFinderTool;shouldKeepLooking()Z", shift = Shift.AFTER, remap = false),
+      locals = LocalCapture.CAPTURE_FAILSOFT,
+      remap = true
+  )
+  public void onGtOreFound(
+      ItemStack itemstack, 
+      World world, Entity entity, 
+      int par4, boolean par5,
+      CallbackInfo ci, 
+      ItemStack searchItem, ItemData data, boolean vanilla, int id, 
+      double cur_x, double cur_y, double cur_z, 
+      int min_x, int min_y, int min_z, 
+      int max_x, int max_y, int max_z, 
+      boolean keepLooking, 
+      int z1, int x1, int y1, 
+      Block tBlock, int meta, ItemStack inWorld, ItemData dataInWorld, List<OrePrefixes> oreTypes
+  ) {
+    if (vanilla || world.isRemote || found < MAX_DAMAGE || !(entity instanceof EntityPlayer)) return;
```

Comment:
Extract `found < MAX_DAMAGE` to `didNotDetectOreCluster()` to better explain comparison reason